### DISCUSSION TITLE
Incomplete backup dump using mysqldump

### DISCUSSION THREAD

**Discussion by yweidmann** [kimai/kimai#3039] [Questions & Answers] [answered]
Dear list

Following the [description](https://www.kimai.org/documentation/backups.html#create-a-database-backup), I was running the following backup-command:
* `mysqldump --single-transaction -u <DatabaseUser> -p -h 127.0.0.1 <Database> /home/.../kimai2_dump.sql;`

The dump-file was created, but it doesn't contain the database-entries of the users. The `<DatabaseUser>` just logs in for administrative purposes. All entries are done by other users (created by using the kimai-web-interface of course).

Question:
* Is there an option missing to dump all entries of all users?

Looking forward to any hints and ideas.

Cheers,
Yvo

**Comment by kevinpapst** [comment, marked as answer]
mysqldump does not know anything about your kimai users. 
you have to use the same database user that is used in your .env file.

**Reply by yweidmann** [reply]
Mh, this is very strange. 
* I checked the `.env` file.
* Running the dump with the user defined in the`.env` file.
* A file was written.

I checked the sql-file carefully. And only one insert-statement is written for each table.

Any ideas, why the dump is not writing all insert-statements for the tables?

**Reply by kevinpapst** [reply]
You get one insert with hundreds of values. 
Unless you define the `--extended-insert` mysqldump will write 500 (I don't know the exact number) rows per insert.
So you will only have multiple inserts if you have more than 500 rows in a table.
This is NOT a Kimai issue: use Google, it works!

**Reply by yweidmann** [reply]
Ahhhhhhhhhhhhhhhh, that was too stupid from my side. I am very sorry about this question.

Works perfectly!

Cheers,
Yvo
